src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -108,6 +109,19 @@ public static class AspireAzureCosmosDBExtensions
         if (serviceKey is null)
         {
             builder.Services.AddSingleton(_ => ConfigureDb());
+
+            var csBuilder = new DbConnectionStringBuilder();
```

Comment:
a) why are we only doing this for the non-EF component?
b) why are we only doing this for the non-Keyed DI path?

Reply:
The PR is incomplete. I was just getting something working to facilitate conversation.